cmake/add_enclave.cmake

```diff
@@ -318,7 +341,7 @@ macro (add_enclave_optee)
   add_dependencies(${ENCLAVE_TARGET} ${ENCLAVE_TARGET}.ld)
   target_include_directories(${ENCLAVE_TARGET}
                              PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/optee)
-  target_link_libraries(${ENCLAVE_TARGET} oeenclave)
+  target_link_libraries(${ENCLAVE_TARGET} oeenclave oecryptombedtls)
```

Comment:
Follow-up to previous comment that got dislocated:

>> Part of the suggestion to introduce the `DEFAULT_TEST_ENCLAVE_CRYPTO_LIB` is so that we have a consistent way to flip the default if we want to encourage OpenSSL as the primary enclave crypto library instead. To that end, we want to avoid hardcoding it in cases where the crypto library is not material to the test.
>> 
>> For OP-TEE, I would like to know that if we swapped the default, we would get testing on the aarch64 configs as to what breaks.
>>
> Given that we don't support OpenSSL on OP-TEE now, what would be the recommendation for the case if we set `DEFAULT_TEST_ENCLAVE_CRYPTO_LIB` to OpenSSL on OPTEE (if we don't do the hardcode here)?

I suggest making the logic from line 199-206 reusable here. My understanding is that you don't have to worry about the variable indicating OpenSSL when building OP-TEE because /CMakeLists.txt already implements the top level blocks for:
- `DEFAULT_TEST_ENCLAVE_CRYPTO_LIB` cannot be OpenSSL without `BUILD_OPENSSL`
- `BUILD_OPENSSL` cannot be specified with `OE_TRUSTZONE`

Having the logic in place here means that you only have to flip the switch on supporting OpenSSL in OP-TEE in one place in the build configuration (root CMakeLists.txt) instead of having to figure out all the other places lower in the tree where that decision was also made locally.

Reply:
Updated as suggested.